src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -174,38 +168,49 @@ def q_impl(
         alpha = self.attrs.get("alpha", 1)
         beta = self.attrs.get("beta", 1)
 
+        # If self.constant_inputs is empty this is an encrypted gemm
+        is_encrypted_gemm = isinstance(self.constant_inputs, dict) and not self.constant_inputs
+
         # If alpha != 1 or beta not in [0, 1], this function must be modified
         assert_true(alpha == 1)
-        assert_true(beta in [0, 1])
+        assert_true(beta in {0, 1})
 
         prepared_inputs = self._prepare_inputs_with_constants(
             *q_inputs, calibrate=False, quantize_actual_values=True
         )
 
-        q_input: QuantizedArray = prepared_inputs[0]
-        q_weights: QuantizedArray = prepared_inputs[1]
-        q_bias: Optional[QuantizedArray] = (
-            None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
-        )
+        q_input = prepared_inputs[0]
+        assert isinstance(q_input, QuantizedArray)
+        q_input2 = prepared_inputs[1]
+        assert isinstance(q_input2, QuantizedArray)
+        q_bias = None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
+        assert isinstance(q_bias, (type(None), QuantizedArray))
```

Comment:
`type(None)` ? that really works ? I'm learning things everyday

Reply:
Looks like it. That's a @fd0r trick.

Reply:
Yes this is needed because python changed its behavior on `NoneType` between versions that we support.
See: https://stackoverflow.com/questions/21095654/what-is-a-nonetype-object

Reply:
got it, but just to make sure, it's strictly identical to having `assert q_bias is None or isinstace(q_bias, QuantizedArray)` right ?

Reply:
yes but a bit more compact imo